### PR TITLE
SID-216

### PR DESCRIPTION
**Pull request by Aliaksandra-Leanovich**

### CODE REVIEW

File: src/components/Button/types.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import { ButtonHTMLAttributes } from "react";
+import { ButtonVariants } from "../../enums";
 
 export interface IButton extends ButtonHTMLAttributes<HTMLButtonElement> {
   handleClick?: () => void;
```
Comment (shumak-yaroslav) [inline src/components/Button/types.ts:5]: VoidFunction

File: src/hooks/use-signin-github.hook.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import { setUserToken } from "../store/slices/userSlice";
+import { app } from "../utils";
+import { getAuthError } from "../helper";
+
+export const useSigninWithGithub = () => {
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const cookies = new Cookies();
+  const localStorageKey = "userToken";
+  const [error, setError] = useState("");
```
Comment (shumak-yaroslav) [inline src/hooks/use-signin-github.hook.ts:22]: useState<string | null>(null);

File: src/hooks/use-signin.hook.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import { setUserToken } from "../store/slices/userSlice";
+import { app } from "../utils";
+
+export const useSignin = (clearErrors: () => void) => {
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const cookies = new Cookies();
+  const localStorageKey = "userToken";
+
+  const [error, setError] = useState("");
```
Comment (shumak-yaroslav) [inline src/hooks/use-signin.hook.ts:23]: useState<string | null>(null);

File: src/components/Button/types.ts (hunk 1 above)
Comment (Aliaksandra-Leanovich) [reply src/components/Button/types.ts:5]: Type '() => void' is not assignable to type '() => VoidFunction'.
  Type 'void' is not assignable to type 'VoidFunction'

Comment (shumak-yaroslav) [reply src/components/Button/types.ts]: () => void -> VoidFunction
(status: string) => void

**Comment by shumak-yaroslav** [comment]
could you please resolve conflicts

**Pull request merged**